### PR TITLE
feat: #340 add new migration deleteFavoritesInWaysAndUsers

### PR DESCRIPTION
**Pull request by dayanych**
Added new migration `deleteFavoritesInWaysAndUsers` and `UsersService` in mw-migrations for fix bug related to favorites.

### CODE REVIEW

File: mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import { doc, writeBatch } from "firebase/firestore";
+import { db } from "../firebase.js";
+import { PATH_TO_USERS_COLLECTION, UsersService } from "../service/UsersService.js";
+import { PATH_TO_WAYS_COLLECTION, WayService } from "../service/WayService.js";
+import { logToFile } from "../utils/logToFile.js";
```
Comment (Ekaterina1994) [inline mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts:5]: I think better to create 2 separate migration files for different models. Also please rename file similar as files in migration folder

Comment (Ekaterina1994) [inline mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts:5]: I think better to create 2 separate migration files for different models. Also please rename file similar as files in migration folder

File: mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    const wayMigrationStartTime = new Date();
+
+    try {
+      log(`started ${way.uuid} migration`);
+
+      const wayRef = doc(db, PATH_TO_WAYS_COLLECTION, way.uuid);
+      batch.set(wayRef, {
+        favoriteForUserUuids: [],
+      });
+
```
Comment (Ekaterina1994) [inline mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts:36]: set method will rewrite all fields in the document

File: mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  for (const user of users) {
+    const userMigrationStartTime = new Date();
+
+    try {
+      log(`started ${user.uuid} migration`);
+
+      const userRef = doc(db,PATH_TO_USERS_COLLECTION, user.uuid);
+      batch.set(userRef, {
+        favoriteWayUuids: [],
+      });
```
Comment (Ekaterina1994) [inline mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts:58]: set method will rewrite all fields in the document

File: mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+  const usersAndWaysMigrationEndTime = new Date();
+  const usersAndWaysMigrationTime = usersAndWaysMigrationEndTime.getTime() - usersAndWaysMigrationStartTime.getTime();
+
+  log(`
+    Migrations report:
+
+    Migration goal:
+    Add createdAt favoriteForUserUuids lastUpdate mentorUuids, deleted monthReportUuids currentMentorUuids
+
```
Comment (Ekaterina1994) [inline mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts:80]: Is it right goal for this migration?

File: mw-migrations/src/service/UsersService.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import { querySnapshotToDTOConverter } from "../converter/querySnapshotToDTOConverter.js";
+import { db } from "../firebase.js";
+import { UserDTOMigration } from "../DTOModel/UserDTO.js";
+
+export const PATH_TO_USERS_COLLECTION = "users";
+
+/**
+ * Provides methods to interact with the Users collection in Firestore.
+ */
+export class UsersService {
```
Comment (Ekaterina1994) [inline mw-migrations/src/service/UsersService.ts:11]: UsersService -> UserService

File: mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts (hunk 2 above)
Comment (dayanych) [reply mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts]: Changed to `update`.

File: mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts (hunk 3 above)
Comment (dayanych) [reply mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts]: Changed to `update`.

File: mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts (hunk 4 above)
Comment (dayanych) [reply mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts:29]: Done.

File: mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts (hunk 1 above)
Comment (dayanych) [reply mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts]: Done.

Comment (dayanych) [reply mw-migrations/src/migrations/deleteFavoritesInWaysAndUsers.ts]: Done.

File: mw-migrations/src/service/UsersService.ts (hunk 5 above)
Comment (dayanych) [reply mw-migrations/src/service/UsersService.ts]: Done.

**Review by Howl404 — APPROVED** [review]

**Review by Ekaterina1994 — APPROVED** [review]

**Pull request merged**